### DISCUSSION TITLE
Use of NextToken for GetOrders

### DISCUSSION THREAD

**Discussion by mahavirbansal** [abuzuhri/Amazon-SP-API-CSharp#581] [Q&A]
I have been sucessfully able to use getorders to download orders. However since the number of orders are too many, i want to split them to smaller size and it appears that "MaxResultsPerPage" would do that and provide with "nexttoken" using which the next page could be retrived and so on. I am able to pass the MaxResultsPerPage however it seems the result is always one single Json page having all the data (even upto 129 although amazon says default is 100) and i never get the nexttoken" tag. 
What am i doing wrong? Can someone shed light on this??

**Comment by TechyChap** [comment]
The library is already doing the paging for you. But if you want to turn that off you can set the MaxNumberOfPages value on the searchOrderList = 1 and you'll get the token back.
